model_external_socket/java/src/org/openda/externalsocket/ExternalSocketModelInstance.java

```diff
@@ -0,0 +1,293 @@
+		String received = socketClient.sendAndReceive(messageIn);
+		System.out.println("Received" + received);
+		String[] split = received.trim().split(";");
+		double[] receivedValues = new double[split.length];
+		for (int i = 0; i < split.length; i++) {
+			double parsedModelResult = Double.parseDouble(split[i]);
+			receivedValues[i] = parsedModelResult;
+		}
+		modelResults = new Vector(receivedValues);
+	}
+
+	public ILocalizationDomains getLocalizationDomains() {
+		throw new RuntimeException("org.openda.externalsocket.ExternalSocketModelInstance.getLocalizationDomains() not implemented yet");
+
+	}
+
+	public IVector[] getObservedLocalization(IObservationDescriptions observationDescriptions, double distance) {
+		throw new RuntimeException("org.openda.externalsocket.ExternalSocketModelInstance.getObservedLocalization() not implemented yet");
+
+	}
+
+	public IVector[] getObservedLocalization(IObservationDescriptions observationDescriptions, double distance, int iDomain) {
+		throw new RuntimeException("org.openda.externalsocket.ExternalSocketModelInstance.getObservedLocalization() not implemented yet");
+
+	}
+
+	public IModelState saveInternalState() {
+		throw new RuntimeException("org.openda.externalsocket.ExternalSocketModelInstance.saveInternalState() not implemented yet");
+
+	}
+
+	public void restoreInternalState(IModelState savedInternalState) {
+		throw new RuntimeException("org.openda.externalsocket.ExternalSocketModelInstance.restoreInternalState() not implemented yet");
+
+	}
+
+	public void releaseInternalState(IModelState savedInternalState) {
+		throw new RuntimeException("org.openda.externalsocket.ExternalSocketModelInstance.releaseInternalState() not implemented yet");
+
+	}
+
+	public IModelState loadPersistentState(File persistentStateFile) {
+		throw new RuntimeException("org.openda.externalsocket.ExternalSocketModelInstance.loadPersistentState() not implemented yet");
+
+	}
+
+	public File getModelRunDir() {
+		return dummyModelDir;
+	}
+
+	public String[] getExchangeItemIDs() {
+		return exchangeItems.keySet().toArray(new String[0]);
+	}
+
+	public String[] getExchangeItemIDs(IPrevExchangeItem.Role role) {
+		throw new RuntimeException("org.openda.externalsocket.ExternalSocketModelInstance.getExchangeItemIDs() not implemented yet");
+
+	}
+
+	public IExchangeItem getDataObjectExchangeItem(String exchangeItemID) {
+		return exchangeItems.get(exchangeItemID);
+
+	}
+
+	public void finish() {
+
+	}
+
+	public void initialize(File workingDir, String[] arguments) {
+		throw new RuntimeException("org.openda.externalsocket.ExternalSocketModelInstance.initialize() not implemented yet");
+
+	}
+
+	public IInstance getParent() {
+		throw new RuntimeException("org.openda.externalsocket.ExternalSocketModelInstance.getParent() not implemented yet");
+
+	}
+
+	public IVector getObservedValues(IObservationDescriptions observationDescriptions) {
+		return modelResults;
+	}
+
+	void sendFinalParameters(double[] stdValues) {
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/OpenDA-Association/OpenDA/pullRequest?prid=6157987)